includes/wikia/cache/TitleBatch.php

```diff
@@ -19,16 +22,17 @@ class TitleBatch {
 	 * @param $titles array List of Title objects or title texts
 	 */
 	public function __construct( $titles ) {
-		foreach ($titles as $k => $title) {
+		$this->titles = [ ];
+		foreach ( $titles as $k => $title ) {
 			if ( $title instanceof Title ) {
-				// nothing to do...
-			} elseif ( is_string($title) ) {
-				$titles[$k] = Title::newFromText($title);
+				$this->titles[ $title->getArticleID() ] = $title;
```

Comment:
This will change the behavior of TitleBatch. Having the following code:
```
$titles = [
  Title::newFromText('NonExistent1'),
  Title::newFromText('NonExistent2'),
];
$titlesFromTitleBatch = (new TitleBatch($titles))->getAll();
```
previously `getAll()` returned both titles, the new version will return only the second one.
Please make sure you don't break anything with this change.